### DISCUSSION TITLE
URI vs URL vs URN

### DISCUSSION THREAD

**Discussion by ahyeon-jung** [learning-with/book-reading-http-reading-perfect-guide#2] [General]
![image](https://github.com/learning-with/book-reading-http-reading-perfect-guide/assets/75254185/aafefa2a-86f5-444c-8e68-404b7a69b83a)
- URI: 특정 리소스를 식별하는 통합 자원 식별자. 웹 기술에서 사용하는 논리적 또는 물리적 리소스를 식별하는 고유한 문자열 시퀀스
  - 위치 정보가 있을 수도 있고 없을 수도 있음
  - 모든 URL은 URI이지만, 프로토콜을 통해 리소스의 위치를 완전히 지정(직접적인 접근 방법이나 위치 정보를 제공)하는 것이 아니면 URL이 아닌 URI
- URL: 리소스 위치를 구분하기 위한 통일된 규약, 약속
  - 프로토콜(스킴), 도메인네임, 포트, 경로, 질의(Query)
  - http는 80, https는 443이 기본 포트 번호로 생략 가능
- URN: 이름을 부여하여 리소스를 식별. 리소스 접근 방법이나 웹에서 위치가 표기되는 방식은 아님 ex. ISBN

---
https://blog.naver.com/wishlan/223311064626

**Comment by EUNCHAEv1006** [comment]
urn 이름만으로 실제 리소스를 찾을 수 있는 방법이 보편화 되지 않았대요
